parameters/src/testnet3/powers.rs

```diff
@@ -529,3 +716,140 @@ impl<E: PairingEngine> ToBytes for PowersOfBetaG<E> {
         self.serialize_with_mode(writer, Compress::No).map_err(|e| e.into())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use snarkvm_curves::bls12_377::Bls12_377;
+
+    #[test]
+    fn test_checked_extensions_of_normal_powers() {
+        let mut powers = PowersOfG::<Bls12_377>::load().unwrap();
+        let incorrect_powers_17 = vec![0u8; NUM_POWERS_17];
+        let powers_17 = impl_download_powers!(NUM_POWERS_17, load_bytes).unwrap();
+        let powers_19 = impl_download_powers!(NUM_POWERS_19, load_bytes).unwrap();
+
+        // Extend powers with the powers available on disk.
+        assert_eq!(powers.num_powers(), NUM_POWERS_15);
+        powers.download_powers_for(0..NUM_POWERS_16).unwrap();
+        assert_eq!(powers.num_powers(), NUM_POWERS_16);
+
+        // Ensure powers can't be extended with incorrect bytes.
+        let result = powers.extend_normal_powers_checked(&incorrect_powers_17, NUM_POWERS_17);
+        assert!(result.is_err());
+        assert_eq!(powers.num_powers(), NUM_POWERS_16);
+
+        // Ensure powers can't be extended with powers of the wrong length.
+        let result = powers.extend_normal_powers_checked(&powers_17, NUM_POWERS_16);
+        assert!(result.is_err());
+        let result = powers.extend_normal_powers_checked(&powers_17, NUM_POWERS_18);
+        assert!(result.is_err());
+        assert_eq!(powers.num_powers(), NUM_POWERS_16);
+
+        // Ensure contiguous powers CAN be extended.
```

Comment:
Is it new behaviour in this PR that we can only extend by the next power?

Reply:
That's a question. There was previously no method for extending powers manually, and so this PR adds that specific behavior. However, at the moment is it legal/prudent to download non-contigous powers?

Reply:
I was just thinking that if we know we want to download `NUM_POWERS_20`, why iteratively download 18 to 19 to 20 (which the code does if I read and recall correctly). On the other hand, I can't imagine this to introduce a huge overhead so this can also be changed later if it ever becomes a performance issue.

Reply:
Yeah, I think you're right about this though if one needs to commit to polynomials with a mix of high and low degree terms and nothing in the middle it's non-sensical to download it in series all the way towards those terms.

Reply:
>  if one needs to commit to polynomials with a mix of high and low degree terms 
Then you need the powers adhering to the biggest polynomial.

The only reason to download the lower powers, is if just downloading the higher power doesn't include that lower power yet.